src/balanced_bst.py

```diff
@@ -0,0 +1,336 @@
+"""Module for Binary Search Tree."""
+
+from queue_ds import Queue
+import timeit
+import random
+
+
+class Node(object):
+    """Node class."""
+
+    def __init__(self, value=None, left=None, right=None):
+        """Init of the Node class."""
+        self.value = value
+        self.left = left
+        self.right = right
+
+
+class BinarySearchTree(object):
+    """Binary Search Tree."""
+
+    """insert(self, val): will insert the value val into the BST. If val is already present, it will be ignored."""
+    """search(self, val): will return the node containing that value, else None"""
+    """size(self): will return the integer size of the BST (equal to the total number of values stored in the tree). It will return 0 if the tree is empty."""
+    """depth(self): will return an integer representing the total number of levels in the tree. If there is one value, the depth should be 1, if two values it will be 2, if three values it may be 2 or three, depending, etc."""
+    """contains(self, val): will return True if val is in the BST, False if not."""
+    """balance(self): will return an integer, positive or negative that represents how well balanced the tree is. Trees which are higher on the left than the right should return a positive value, trees which are higher on the right than the left should return a negative value. An ideally-balanced tree should return 0."""
+    """in_order(self): will return a generator that will return the values in the tree using in-order traversal, one at a time."""
+    """pre_order(self): will return a generator that will return the values in the tree using pre-order traversal, one at a time."""
+    """post_order(self): will return a generator that will return the values in the tree using post_order traversal, one at a time."""
+    """breadth_first(self): will return a generator that will return the values in the tree using breadth-first traversal, one at a time."""
+
```

Comment:
You might be missing a method here.   Also you don't need to wrap every line in the """, just once at the beginning and once and once at the end.